### PR TITLE
net: add IP.IsPrivate

### PR DESCRIPTION
**Pull request by 6543**
Adds a new method IsPrivate to check if an IP is
private according to RFC 1918 & RFC 4193.

Fixes #29146

### CODE REVIEW

**Comment by 6543** [comment]
creddits to @MaxSem as I used your pull (#30278) as template :)

I would mention you in the comment but it does not look like souch information is wantend in the commit history so I leave it out...

**Comment by 6543** [comment]
@googlebot I signed it!

**Comment by bl-ue** [comment]
Any progress on the fantastic feature??

**Comment by 6543** [comment]
@bl-ue feature-freeze will go into 1.17

**Comment by bl-ue** [comment]
Thank you @6543!  I love this feature and can't wait for 1.17.

File: src/net/ip_test.go [hunk 1]
```diff
@@ -690,6 +690,25 @@ var ipAddrScopeTests = []struct {
...
+	{IP.IsPrivate, IPv4(1, 1, 1, 1), false},
+	{IP.IsPrivate, IPv4(9, 255, 255, 255), false},
+	{IP.IsPrivate, IPv4(10, 0, 0, 0), true},
+	{IP.IsPrivate, IPv4(10, 255, 255, 255), true},
+	{IP.IsPrivate, IPv4(11, 0, 0, 0), false},
+	{IP.IsPrivate, IPv4(172, 15, 255, 255), false},
+	{IP.IsPrivate, IPv4(172, 16, 0, 0), true},
+	{IP.IsPrivate, IPv4(172, 16, 255, 255), true},
+	{IP.IsPrivate, IPv4(172, 32, 0, 0), false},
+	{IP.IsPrivate, IPv4(192, 167, 255, 255), true},
```
Comment (sapk) [inline src/net/ip_test.go:703]: This test should be false:
```suggestion
	{IP.IsPrivate, IPv4(192, 167, 255, 255), false},
```

Comment (sapk) [reply src/net/ip_test.go:703]: The new func work as intended. It just the test that need to be updated.

Comment (6543) [reply src/net/ip_test.go:703]: look like i got a copy-paste issue ... fixing ...

**Comment by odeke-em** [comment]
@6543 please take a look at, and address the feedback that I provided, then we’ll be ready to go. Thank you.

**Comment by 6543** [comment]
just had no time to allocate it to this :)
give me few minutes :)

**Comment by 6543** [comment]
@odeke-em hope, I got it right this time :rocket:

**Comment by odeke-em** [comment]
@6543 you got it, congrats and thank you!

**Comment by bl-ue** [comment]
🎊 🎉 🚀

**Pull request closed without merging**